// @uppy/components/lib/Dropzone.js
/* eslint-disable jsx-a11y/click-events-have-key-events */
import { h } from 'preact';
import { useState, useRef } from 'preact/hooks';
import { clsx } from 'clsx';
export default function Dropzone(props) {
  const fileInputRef = useRef(null);
  const [isDragging, setIsDragging] = useState(() => false);
  const {
    width,
    height,
    note,
    noClick,
    ctx
  } = props;
  function handleDrop(event) {
    var _event$dataTransfer, _ctx$uppy;
    event.preventDefault();
    event.stopPropagation();
    setIsDragging(false);
    const files = Array.from(((_event$dataTransfer = event.dataTransfer) == null ? void 0 : _event$dataTransfer.files) || []);
    if (!files.length) return;
    (_ctx$uppy = ctx.uppy) == null || _ctx$uppy.addFiles(files.map(file => ({
      source: 'drag-drop',
      name: file.name,
      type: file.type,
      data: file
    })));
  }
  function handleDragOver(event) {
    event.preventDefault();
    event.stopPropagation();
  }
  function handleDragEnter(event) {
    event.preventDefault();
    event.stopPropagation();
    setIsDragging(true);
  }
  function handleDragLeave(event) {
    event.preventDefault();
    event.stopPropagation();
    setIsDragging(false);
  }
  function handleClick() {
    var _fileInputRef$current;
    if (noClick) return;
    (_fileInputRef$current = fileInputRef.current) == null || _fileInputRef$current.click();
  }
  function handleFileInputChange(event) {
    var _ctx$uppy2;
    const input = event.target;
    const files = Array.from(input.files || []);
    if (!files.length) return;
    (_ctx$uppy2 = ctx.uppy) == null || _ctx$uppy2.addFiles(files.map(file => ({
      source: 'drag-drop',
      name: file.name,
      type: file.type,
      data: file
    })));

    // Reset the input value so the same file can be selected again
    input.value = '';
  }
  return h("div", {
    className: "uppy-reset",
    "data-uppy-element": "dropzone"
  }, h("input", {
    type: "file",
    className: "uppy:hidden",
    ref: fileInputRef,
    multiple: true,
    onChange: handleFileInputChange
  }), h("div", {
    role: "button",
    tabIndex: 0,
    style: {
      width: width || '100%',
      height: height || '100%'
    },
    className: clsx('uppy:border-2 uppy:border-dashed uppy:border-gray-300', 'uppy:rounded-lg uppy:p-6 uppy:bg-gray-50', 'uppy:transition-colors uppy:duration-200', {
      'uppy:bg-blue-50': isDragging
    }, {
      'uppy:cursor-pointer uppy:hover:bg-blue-50': !noClick
    }),
    onDrop: handleDrop,
    onDragOver: handleDragOver,
    onDragEnter: handleDragEnter,
    onDragLeave: handleDragLeave,
    onClick: handleClick
  }, h("div", {
    className: "uppy:flex uppy:flex-col uppy:items-center uppy:justify-center uppy:h-full uppy:space-y-3"
  }, h("p", {
    className: "uppy:text-gray-600"
  }, "Drop files here or click to add them")), note ? h("div", {
    className: "uppy:text-sm uppy:text-gray-500"
  }, note) : null));
}
