// @uppy/components/lib/FilesList.js
/* eslint-disable react/destructuring-assignment */
import { Fragment, h } from 'preact';
import { useState, useEffect } from 'preact/hooks';
import prettyBytes from 'pretty-bytes';
import { clsx } from 'clsx';
import { Thumbnail } from './index.js';
export default function FilesList(props) {
  const [files, setFiles] = useState(() => []);
  const {
    ctx,
    editFile
  } = props;
  useEffect(() => {
    var _ctx$uppy;
    const onStateUpdate = (prev, next, patch) => {
      if (patch != null && patch.files) {
        setFiles(Object.values(patch.files));
      }
    };
    (_ctx$uppy = ctx.uppy) == null || _ctx$uppy.on('state-update', onStateUpdate);
    return () => {
      var _ctx$uppy2;
      (_ctx$uppy2 = ctx.uppy) == null || _ctx$uppy2.off('state-update', onStateUpdate);
    };
  }, [ctx.uppy]);
  return h("ul", {
    "data-uppy-element": "files-list",
    className: "uppy-reset uppy:my-4"
  }, files == null ? void 0 : files.map(file => {
    var _file$progress, _file$progress2, _file$progress3;
    return h("li", {
      key: file.id
    }, h(Fragment, null, h("div", {
      className: "uppy:flex uppy:items-center uppy:gap-2"
    }, h("div", {
      className: "uppy:w-[32px] uppy:h-[32px]"
    }, h(Thumbnail, {
      width: "32px",
      height: "32px",
      file: file
    })), h("p", {
      className: "uppy:truncate"
    }, file.name), h("p", {
      className: "uppy:text-gray-500 uppy:tabular-nums uppy:min-w-18 uppy:text-right uppy:ml-auto"
    }, prettyBytes(file.size || 0)), h(Fragment, null, editFile && h("button", {
      type: "button",
      className: "uppy:flex uppy:rounded uppy:text-blue-500 uppy:hover:text-blue-700 uppy:bg-transparent uppy:transition-colors",
      onClick: () => {
        editFile(file);
      }
    }, "edit"), h("button", {
      type: "button",
      className: "uppy:flex uppy:rounded uppy:text-blue-500 uppy:hover:text-blue-700 uppy:bg-transparent uppy:transition-colors",
      onClick: () => {
        var _ctx$uppy3;
        (_ctx$uppy3 = ctx.uppy) == null || _ctx$uppy3.removeFile(file.id);
      }
    }, "remove"))), h("progress", {
      max: "100",
      className: clsx('uppy:w-full uppy:h-[2px] uppy:appearance-none uppy:bg-gray-100 uppy:rounded-full uppy:overflow-hidden uppy:[&::-webkit-progress-bar]:bg-gray-100 uppy:block uppy:my-2', {
        'uppy:[&::-webkit-progress-value]:bg-green-500 uppy:[&::-moz-progress-bar]:bg-green-500': (_file$progress = file.progress) == null ? void 0 : _file$progress.uploadComplete,
        'uppy:[&::-webkit-progress-value]:bg-red-500 uppy:[&::-moz-progress-bar]:bg-red-500': file.error,
        'uppy:[&::-webkit-progress-value]:bg-blue-500 uppy:[&::-moz-progress-bar]:bg-blue-500': !((_file$progress2 = file.progress) != null && _file$progress2.uploadComplete) && !file.error
      }),
      value: ((_file$progress3 = file.progress) == null ? void 0 : _file$progress3.percentage) || 0
    })));
  }));
}
