// @uppy/components/lib/FilesGrid.js
/* eslint-disable react/destructuring-assignment */
import { Fragment, h } from 'preact';
import { useState, useEffect } from 'preact/hooks';
import prettyBytes from 'pretty-bytes';
import { clsx } from 'clsx';
import { Thumbnail } from './index.js';
export default function FilesGrid(props) {
  const [files, setFiles] = useState(() => []);
  const {
    ctx,
    editFile
  } = props;
  function gridColsClass() {
    return {
      1: 'uppy:grid-cols-1',
      2: 'uppy:grid-cols-2',
      3: 'uppy:grid-cols-3',
      4: 'uppy:grid-cols-4',
      5: 'uppy:grid-cols-5',
      6: 'uppy:grid-cols-6'
    }[props.columns || 2] || 'uppy:grid-cols-2';
  }
  useEffect(() => {
    var _ctx$uppy;
    const onStateUpdate = (prev, next, patch) => {
      if (patch != null && patch.files) {
        setFiles(Object.values(patch.files));
      }
    };
    (_ctx$uppy = ctx.uppy) == null || _ctx$uppy.on('state-update', onStateUpdate);
    return () => {
      var _ctx$uppy2;
      (_ctx$uppy2 = ctx.uppy) == null || _ctx$uppy2.off('state-update', onStateUpdate);
    };
  }, [ctx.uppy]);
  return h("div", {
    "data-uppy-element": "files-grid",
    className: clsx('uppy:reset uppy:my-4 uppy:grid uppy:gap-4', gridColsClass())
  }, files == null ? void 0 : files.map(file => h("div", {
    className: "uppy:flex uppy:flex-col uppy:items-center uppy:gap-2",
    key: file.id
  }, h(Fragment, null, h(Thumbnail, {
    images: true,
    file: file
  }), h("div", {
    className: "uppy:w-full uppy-reset"
  }, h("p", {
    className: "uppy:font-medium uppy:truncate",
    title: file.name
  }, file.name), h("div", {
    className: "uppy:flex uppy:items-center uppy:gap-2"
  }, h("p", {
    className: " uppy:text-gray-500 uppy:tabular-nums "
  }, prettyBytes(file.size || 0)), editFile && h("button", {
    type: "button",
    className: "uppy:flex uppy:rounded uppy:text-blue-500 uppy:hover:text-blue-700 uppy:bg-transparent uppy:transition-colors",
    onClick: () => {
      editFile(file);
    }
  }, "edit"), h("button", {
    type: "button",
    className: "uppy:flex uppy:rounded uppy:text-blue-500 uppy:hover:text-blue-700 uppy:bg-transparent uppy:transition-colors",
    onClick: () => {
      var _ctx$uppy3;
      (_ctx$uppy3 = ctx.uppy) == null || _ctx$uppy3.removeFile(file.id);
    }
  }, "remove")))))));
}
