// @uppy/vue/lib/headless/context-provider.js
import { defineComponent, provide, reactive, ref, onMounted, onBeforeUnmount } from 'vue';
import { createUppyEventAdapter } from '@uppy/components';
export const UppyContextSymbol = Symbol('uppy');
export const UppyContextProvider = defineComponent({
  name: 'UppyContextProvider',
  props: {
    uppy: {
      type: Object,
      required: true
    }
  },
  setup(props, _ref) {
    let {
      slots
    } = _ref;
    const status = ref('init');
    const progress = ref(0);
    const uppyContext = reactive({
      uppy: props.uppy,
      status: 'init',
      progress: 0
    });

    // Provide the context immediately instead of in onMounted
    provide(UppyContextSymbol, uppyContext);
    let uppyEventAdapter;
    onMounted(() => {
      if (!props.uppy) {
        throw new Error('UppyContextProvider: passing `uppy` as a prop is required');
      }
      uppyEventAdapter = createUppyEventAdapter({
        uppy: props.uppy,
        onStatusChange: newStatus => {
          status.value = newStatus;
          uppyContext.status = newStatus;
        },
        onProgressChange: newProgress => {
          progress.value = newProgress;
          uppyContext.progress = newProgress;
        }
      });
    });
    onBeforeUnmount(() => {
      var _uppyEventAdapter;
      (_uppyEventAdapter = uppyEventAdapter) == null || _uppyEventAdapter.cleanup();
    });
    return () => slots.default == null ? void 0 : slots.default();
  }
});
export default UppyContextProvider;
